iceoryx_hoofs/source/posix_wrapper/thread.cpp

```diff
@@ -16,38 +16,97 @@
 
 #include "iceoryx_hoofs/posix_wrapper/thread.hpp"
 #include "iceoryx_hoofs/cxx/helplets.hpp"
+#include "iceoryx_hoofs/internal/log/hoofs_logging.hpp"
 #include "iceoryx_hoofs/posix_wrapper/posix_call.hpp"
 
 namespace iox
 {
 namespace posix
 {
-void setThreadName(pthread_t thread, const ThreadName_t& name) noexcept
+Thread::~Thread() noexcept
 {
-    posixCall(iox_pthread_setname_np)(thread, name.c_str()).successReturnValue(0).evaluate().or_else([](auto& r) {
-        // String length limit is ensured through cxx::string
-        // ERANGE (string too long) intentionally not handled to avoid untestable and dead code
-        std::cerr << "This should never happen! " << r.getHumanReadableErrnum() << std::endl;
-        cxx::Ensures(false && "internal logic error");
-    });
+    if (m_isJoinable)
+    {
+        auto joinResult = posixCall(pthread_join)(m_threadHandle, nullptr).successReturnValue(0).evaluate();
+        if (joinResult.has_error())
+        {
+            switch (joinResult.get_error().errnum)
+            {
+            case EDEADLK:
+                LogError() << "A deadlock was detected when attempting to join the thread.";
```

Comment:
I guess the thread would keep running ... or maybe keep deadlocking, right?

The philosophical question would be if we should call the error handler? Worst case one could get out of memory if this happens repeatedly.